.github/workflows/kafka_demo_test.yml

```diff
@@ -123,7 +123,7 @@ jobs:
     name: Test Workload 
     uses: ./.github/workflows/_test_workload.yml
     needs: [push_images, create-aks-cluster, deploy-kafka-test]
-    if: always() && (needs.push_images.result == 'success' || needs.push_images.result == 'skipped')
```

Comment:
@hgarvison 
I want to change this but don't want to cause any unexpected failures. Why did we want to run this step when push image result is skipped? Because when the pipeline is run from main and push image step is skipped, this step fails due to image DNE.

Reply:
The push image step should only be ran during PRs and skipped otherwise. So the Test Workload step should run with the created image during PRs or a default image defined in the _test_workload.yml otherwise.

Reply:
Sorry I put the question on the wrong line. So the question should be for line 133 for the cleanup step. The pipeline is run from main and push image step is skipped; this step fails due to image DNE.

Reply:
oh yeah that was copy paste, we should prob get rid of the skipped part